tests/unit/SettingsTest.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace Acquia\Drupal\RecommendedSettings\Tests;
+
+use Acquia\Drupal\RecommendedSettings\Helpers\Filesystem as DrsFilesystem;
+use Acquia\Drupal\RecommendedSettings\Settings;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Filesystem\Filesystem;
+
+class SettingsTest extends TestCase {
+
+  /**
+   * The recommended settings object.
+   *
+   * @var string
+   */
+  protected $settings;
+
+  /**
+   * The path to drupal webroot directory.
+   *
+   * @var string
+   */
+  protected $drupalRoot;
+
+  /**
+   * The symfony file-system object.
+   *
+   * @var \Symfony\Component\Filesystem\Filesystem
+   */
+  protected Filesystem $fileSystem;
+
+  /**
+   * The symfony file-system object.
+   *
+   * @var \Acquia\Drupal\RecommendedSettings\Helpers\Filesystem
+   */
+  protected DrsFilesystem $drsFileSystem;
+
+  /**
+   * Set up test environmemt.
+   */
+  public function setUp(): void {
+    $this->drupalRoot = dirname(__FILE__);
+    $docroot = $this->drupalRoot . '/docroot';
+    $this->drsFileSystem = new DrsFilesystem();
+    $this->drsFileSystem->ensureDirectoryExists($docroot . '/sites/default');
+    $this->fileSystem = new Filesystem();
+    $this->fileSystem->touch($docroot . '/sites/default/default.settings.php');
+    $this->settings = new Settings($docroot, "default");
+    $this->settings->generate();
```

Comment:
Pass db_spec here like
`$this->settings->generate();`

Reply:
Incorporated.